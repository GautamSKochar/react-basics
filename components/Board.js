import React from 'react';
import Featured from './Features';
import { connect } from 'react-redux';
import { addComment, editComment, deleteComment } from '../actions/commentActions';

class Board extends React.Component {

    addNew(text) {
        this.props.addComment(text);
    }

    removeComment(i) {
        this.props.deleteComment(i);
    }

    updateComment(newText, i) {
        this.props.editComment(i, newText);
    }

    eachComment(txt, i) {
        return (
            <Featured key={i} index={i}
                updateCommentText={this.updateComment.bind(this)}
                deleteFromBoard={this.removeComment.bind(this)}>{txt}</Featured>
        );
    }

    render() {
        return (
            <div className="container" style={{ float: 'left', display: 'inline-block', whiteSpace: 'nowrap' }}>
                <h3>Board of comments</h3>
                <hr />
                <button onClick={() => this.addNew("newww text")} className="btn btn-info">Add New</button>
                <hr />
                {this.props.comments.map(this.eachComment.bind(this))}
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        comments: state.comments.comments
    };
}

const mapDispatchToPropss = (dispatch) => {
    return {
        addComment: (newText) => {
            dispatch(addComment(newText))
        },
        editComment: (id, newText) => {
            dispatch(editComment(id, newText))
        },
        deleteComment: (id) => {
            dispatch(deleteComment(id))
        }
    };
}

export default connect(mapStateToProps, mapDispatchToPropss)(Board);