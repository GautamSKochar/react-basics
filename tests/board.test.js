import React from 'react';
import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import Board from '../components/Board';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

const raf = global.requestAnimationFrame = (cb) => {
    setTimeout(cb, 0)
}
configure({ adapter: new Adapter() });

test('Board Component Should render as expected', () => {
    const component = shallow(<Board />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
    //const wrapper1 = mount(<Board />);
    //wrapper1.find('button').simulate('click');
    // expect(wrapper.find('.ReactCodeMirror')).toHaveLength(1);
    //const btn = shallow(<button onClick={() => this.addNew("newww text")} className="btn btn-info">Add New</button>);
    const value = 'newww text';
    const onClick = jest.fn();
    const wrapper = shallow(
        <button onClick={() => onClick} className="btn btn-info">Add New</button>
    );

    expect(wrapper).toMatchSnapshot();
        wrapper.find('button').simulate('onClick', {
        target: { value },
    });
    // expect(onClick).toBeCalledWith(value);
});    