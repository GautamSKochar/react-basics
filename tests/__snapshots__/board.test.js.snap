// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board Component Should render as expected 1`] = `
<div
  className="container"
  style={
    Object {
      "display": "inline-block",
      "float": "left",
      "whiteSpace": "nowrap",
    }
  }
>
  <h3>
    Board of comments
  </h3>
  <hr />
  <button
    className="btn btn-info"
    onClick={[Function]}
  >
    Add New
  </button>
  <hr />
  <Featured
    deleteFromBoard={[Function]}
    index={0}
    key="0"
    updateCommentText={[Function]}
  >
    This is my new note
  </Featured>
  <Featured
    deleteFromBoard={[Function]}
    index={1}
    key="1"
    updateCommentText={[Function]}
  >
    this is another note comment
  </Featured>
  <Featured
    deleteFromBoard={[Function]}
    index={2}
    key="2"
    updateCommentText={[Function]}
  >
    We have enough comments
  </Featured>
</div>
`;

exports[`Board Component Should render as expected 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <button
    className="btn btn-info"
    onClick={[Function]}
  >
    Add New
  </button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Add New",
      "className": "btn btn-info",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": "Add New",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Add New",
        "className": "btn btn-info",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "Add New",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
